src/main/scala/gen/gmos/GmosLongslitProblem.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic
+package gen.gmos
+
+import basic.enum.ImageQuality
+import gem.enum.GmosXBinning
+
+import fs2._
+
+/**
+ * Possible issues that may arise as a sequence is calculated.
+ */
+sealed trait GmosLongslitProblem extends Product with Serializable {
+  def message: String
+
+  def exception: GmosLongslitException =
+    GmosLongslitException(this)
+
+  def raiseError[F[_]: RaiseThrowable]: Stream[F, INothing] =
+    Stream.raiseError(exception)
```

Comment:
The only way that I know of to report a problem and stop the `Stream` evaluation is to raise an exception.  So I created a `GmosLongslitException` just to wrap an ADT describing the problem.  Is this the correct way to go about this?

Reply:
You can make a `Stream[Option[A]]` or `Stream[Either[E, A]]` and there are methods that will terminate the stream if you push a `None` or a `Left`. I don't know if that is applicable for your case but I use that on the seqexec for graceful termination

Reply:
Thanks, a stream of `Either` might be the correct way to go.  It sort of complicates the API but it does make it clearer what might happen and force the caller to deal with it.